python/lsst/pipe/tasks/hips.py

```diff
@@ -936,16 +1003,20 @@ def run(self, bands, max_order, hips_exposure_handle_dict, do_color=False):
                 Q=self.config.png_gray_asinh_softening,
             )
         else:
-            png_color_mapping = AsinhMapping(
-                self.config.png_color_asinh_minimum,
-                self.config.png_color_asinh_stretch,
-                Q=self.config.png_color_asinh_softening,
-            )
+            match self.config.rgbStyle:
+                case "lupton":
+                    png_color_mapping = AsinhMapping(
+                        self.config.png_color_asinh_minimum,
+                        self.config.png_color_asinh_stretch,
+                        Q=self.config.png_color_asinh_softening,
+                    )
 
-            bcb = self.config.blue_channel_band
-            gcb = self.config.green_channel_band
-            rcb = self.config.red_channel_band
-            colorstr = f"{bcb}{gcb}{rcb}"
+                    bcb = self.config.blue_channel_band
+                    gcb = self.config.green_channel_band
+                    rcb = self.config.red_channel_band
+                    colorstr = f"{bcb}{gcb}{rcb}"
```

Comment:
I'm guessing this is a lot of work, but it would be really nice if the `lupton` case could just be a different `rgbGenerator` implementation, with that `ConfigurableField` maybe replaced with a `RegistryField`.

Is the goal to deprecate the `lupton` version shortly?  If not, do you need to un-deprecate any of its config parameters?

Reply:
Yeah, the goal is to keep Lipton only in the short term as we transition to the new framework, so it is preserved mostly as is.